cpp/include/kvikio/defaults.hpp

```diff
@@ -56,6 +58,64 @@ CompatMode getenv_or(std::string_view env_var_name, CompatMode default_val);
 template <>
 std::vector<int> getenv_or(std::string_view env_var_name, std::vector<int> default_val);
 
+/**
+ * @brief Get the environment variable value from a candidate list
+ *
+ * @tparam T Type of the environment variable value
+ * @param env_var_names Name of the environment variable
+ * @param default_val Default value of the environment variable, if none of the candidates has been
+ * found
+ * @return A tuple of (`env_var_name`, `result`, `has_found`), where:
+ *   - If the environment variable is not set by any of the candidates, `has_found` will be false,
+ * `result` be `default_val`, and `env_var_name` be empty.
+ *   - If the environment variable is set by `env_var_name`, then `has_found` will be true, and
+ * `result` be the set value.
+ *
+ * @throws std::invalid_argument if:
+ *   - `env_var_names` is empty
+ *   - An environment variable from `env_var_names` has an empty value, e.g. by setting `export
+ * KVIKIO_NTHREADS=`
+ *   - Multiple candidates have been set at the same time
+ *   - An invalid value is given
+ */
+template <typename T>
+std::tuple<std::string_view, T, bool> getenv_or(
+  std::initializer_list<std::string_view> env_var_names, T default_val)
+{
+  KVIKIO_NVTX_FUNC_RANGE();
```

Comment:
Do we need this range? reading env vars is usually very fast, I don't see when having this range would help with profiling.